### PR TITLE
feat: column operations actions

### LINKED ISSUE #915: [BUG]: delete regular board with team
## Description

As an  user that is a superAdmin, responsible or team admin of a regular board, I should be able to delete that board.

### Steps to reproduce

Create a regular board with a team as participants and then try to delete the board. It gives an error. 

### Expected result

The board should be deleted.

### Actual result

I only can delete successfully a split and a regular board with participants. If it is a regular board of a team, it gives me an error.

### PR DESCRIPTION
**Pull request by CatiaAntunes96**
Relates to #881 
Fixes: #915 

## Proposed Changes

  - On the backend add updateColumn function, to update title, color and text card on a column
  - Fix delete regular board with a team as participants 

This pull request closes #915

### CODE REVIEW

File: backend/src/modules/boards/controller/boards.controller.ts [hunk 1]
```diff
@@ -312,4 +316,46 @@ export default class BoardsController {
...
+	async updateColumn(@Param() { boardId }: BaseParam, @Body() columnData: UpdateColumnDto) {
+		const board = await this.updateBoardApp.updateColumn(boardId, columnData);
+
+		if (!board) throw new BadRequestException(UPDATE_FAILED);
+
+		if (columnData.socketId) {
+			this.socketService.sendUpdatedBoard(boardId, columnData.socketId);
+		}
+
+		return board;
```
Comment (rpvsilva) [inline backend/src/modules/boards/controller/boards.controller.ts:359]: You should not do this kind of logic on controller side

File: backend/src/modules/boards/services/update.board.service.ts [hunk 2]
```diff
@@ -323,4 +325,43 @@ export default class UpdateBoardServiceImpl implements UpdateBoardServiceInterfa
...
+					arrayFilters: [{ 'column._id': column._id }],
+					new: true
+				}
+			)
+			.populate(BoardDataPopulate)
+			.lean()
+			.exec();
+
+		return result;
+	}
```
Comment (rpvsilva) [inline backend/src/modules/boards/services/update.board.service.ts:353]:
```suggestion
	updateColumn(boardId: string, column: UpdateColumnDto) {
		return this.boardModel
			.findOneAndUpdate(
				{
					_id: boardId,
					'columns._id': column._id
				},
				{
					$set: {
						'columns.$[column].color': column.color,
						'columns.$[column].title': column.title,
						'columns.$[column].cardText': column.cardText
					}
				},
				{
					arrayFilters: [{ 'column._id': column._id }],
					new: true
				}
			)
			.populate(BoardDataPopulate)
			.lean()
			.exec();
	}
```

File: backend/src/modules/boards/services/update.board.service.ts [hunk 3]
```diff
@@ -323,4 +325,43 @@ export default class UpdateBoardServiceImpl implements UpdateBoardServiceInterfa
...
+			.find({
+				_id: boardId,
+				'columns._id': columnId
+			})
+			.populate(BoardDataPopulate)
+			.lean()
+			.exec();
+
+		return result;
+	}
```
Comment (rpvsilva) [inline backend/src/modules/boards/services/update.board.service.ts:366]:
```suggestion
	emptyColumn(boardId: string, columnId: string) {
		return this.boardModel
			.find({
				_id: boardId,
				'columns._id': columnId
			})
			.populate(BoardDataPopulate)
			.lean()
			.exec();
	}
```

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: backend/src/modules/boards/controller/boards.controller.ts [hunk 4]
```diff
@@ -312,4 +316,38 @@ export default class BoardsController {
...
+		description: 'Forbidden',
+		type: ForbiddenResponse
+	})
+	@ApiInternalServerErrorResponse({
+		description: 'Internal Server Error',
+		type: InternalServerErrorResponse
+	})
+	@BoardUser([BoardRoles.RESPONSIBLE, TeamRoles.ADMIN, TeamRoles.STAKEHOLDER])
+	@UseGuards(BoardUserGuard)
+	@Put(':boardId/column')
```
Comment (nunocaseiro) [inline backend/src/modules/boards/controller/boards.controller.ts:349]:
```suggestion
	@Put(':boardId/column/:columnId')
```
I know you are sending the columnId in the dto but I think the columnId should be included in the route as I'm not mistaken it is a best practice

**Review by nunocaseiro — APPROVED** [review]

**Review by nunocaseiro — APPROVED** [review]

**Review by rpvsilva — APPROVED** [review]
LGTM 🔥

**Pull request merged**